### PR TITLE
Add new ML cluster settings for 2.5

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Signed-off-by: Naarcha-AWS <naarcha@amazon.com>

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/cluster-settings.md [hunk 1]
```diff
@@ -12,32 +12,30 @@ This page provides an overview of `opensearch.yml` settings that can be configur
...
 plugins.ml_commons.only_run_on_ml_node: true
 ```
 
-### Description
-
-If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as "false" on production cluster. 
-
 ### Values
 
 - Default value: `false`
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:25]:
```suggestion
- Default value: `true`
```

File: _ml-commons-plugin/cluster-settings.md [hunk 2]
```diff
@@ -47,49 +45,44 @@ plugins.ml_commons.task_dispatch_policy: round_robin
...
-### Description
-
-When returning runtime information with the [profile API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api#profile), ML Commons will run a regular sync up job to sync up newly loaded or unloaded models on each node. When set to `0`, ML Commons immediately stops sync up jobs.
-
 ### Values
 
 - Default value: `10`
 - Value range: [0, 86,400]
 
 ## Predict monitoring requests
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:61]:
```suggestion
## Monitoring predict requests
```

File: _ml-commons-plugin/cluster-settings.md [hunk 3]
```diff
@@ -47,49 +45,44 @@ plugins.ml_commons.task_dispatch_policy: round_robin
...
 
+Controls how many upload model tasks can run in parallel on one node. If set to `0`, you cannot upload models to any node.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/cluster-settings.md:63]:
```suggestion
Controls how many predict request will be monitored on one node. If set to `0`, will clean up all monitoring predict request in cache and will not monitor new predict request any more. 
```

File: _ml-commons-plugin/cluster-settings.md (hunk 2 above)
Comment (Naarcha-AWS) [reply _ml-commons-plugin/cluster-settings.md:61]: Is there a reason why monitoring has to appear first? This is meant to be in line with our style and standard for headings?

File: _ml-commons-plugin/cluster-settings.md (hunk 3 above)
Comment (Naarcha-AWS) [inline _ml-commons-plugin/cluster-settings.md:63]:
```suggestion
Controls how many predict requests are monitored on one node. If set to `0`, OpenSearch clears all monitoring predict requests in the node's cache, and does not monitor predict requests from that point forward.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 4]
```diff
@@ -12,32 +12,30 @@ This page provides an overview of `opensearch.yml` settings that can be configur
 
 ## Run tasks and models on ML nodes only
 
+If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as "false" on production cluster.
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/cluster-settings.md:15]:
```suggestion
If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as `false` on a production cluster. 
```

File: _ml-commons-plugin/cluster-settings.md [hunk 5]
```diff
@@ -12,32 +12,30 @@ This page provides an overview of `opensearch.yml` settings that can be configur
...
-If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as "false" on production cluster. 
-
 ### Values
 
 - Default value: `false`
 - Value range: `true` or `false`
 
 ## Dispatch tasks to ML node 
 
+`round_robin` dispatches ML tasks to ML nodes using round robin routing. `least_load` gathers all an ML nodes' runtime information, like JVM heap memory usage and running tasks, then dispatches tasks to the ML node with the least load.
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/cluster-settings.md:30]:
```suggestion
`round_robin` dispatches ML tasks to ML nodes using round robin routing. `least_load` gathers all ML nodes' runtime information, such as JVM heap memory usage and running tasks, then dispatches tasks to the ML node with the least load.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 6]
```diff
@@ -99,33 +92,30 @@ Controls how many upload model tasks can run in parallel on one node. If set to
 
 ## Load model tasks per node
 
+Controls how many load model tasks can run in parallel on one node. If set as 0, you cannot load models to any node.
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/cluster-settings.md:95]:
```suggestion
Controls how many load model tasks can run in parallel on one node. If set to `0`, you cannot load models to any node.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 7]
```diff
@@ -99,33 +92,30 @@ Controls how many upload model tasks can run in parallel on one node. If set to
...
-Controls how many load model tasks can run in parallel on one node. If set as 0, you cannot load models to any node.
-
 ### Values 
 
 - Default value: `10`
 - Value range: [0, 10]
 
 ## Add trusted URL
 
+The default value allows uploading a model file from any http/https/ftp/local file. You can change this value to restrict trusted model URL.
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/cluster-settings.md:110]: Should it be "trusted model URLs"?
```suggestion
The default value allows uploading a model file from any `http`, `https`, `ftp`, or local file. You can change this value to restrict trusted model URL.
```

File: _ml-commons-plugin/model-serving-framework.md [hunk 8]
```diff
@@ -25,10 +25,12 @@ As of OpenSearch 2.4, the model-serving framework only supports text embedding m
 
 ### Model format
 
-To use a model in OpenSearch, you'll need to export the model into a portable format. As of Version 2.4, OpenSearch only supports the [TorchScript](https://pytorch.org/docs/stable/jit.html) format.
+To use a model in OpenSearch, you'll need to export the model into a portable format. As of Version 2.5, OpenSearch only supports the [TorchScript](https://pytorch.org/docs/stable/jit.html) format and [ONNX][https://onnx.ai/] formats.
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/model-serving-framework.md:28]:
```suggestion
To use a model in OpenSearch, you'll need to export the model into a portable format. As of Version 2.5, OpenSearch only supports the [TorchScript](https://pytorch.org/docs/stable/jit.html) and [ONNX][https://onnx.ai/] formats.
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Left suggestions.

**Review by hdhalter — COMMENTED** [review]
Instead of "This page provides an overview of `opensearch.yml` settings that can be configured for the ML commons plugin.", can we start with something like, "You can modify settings for the ML commons plugin in the 'opensearch.yml' file." I just want to get away from "This section, page, etc. includes..." as an introductory sentence. Thanks!

File: _ml-commons-plugin/cluster-settings.md [hunk 9]
```diff
@@ -7,37 +7,35 @@ nav_order: 10
 
 # ML Commons cluster settings
 
-This page provides an overview of `opensearch.yml` settings that can be configured for the ML commons plugin.
+To enhance and customize your OpenSearch cluster for machine learning (ML), you can add any of the following settings for the ML commons plugin in your 'opensearch.yml' file.
 
 
 ## Run tasks and models on ML nodes only
 
+If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as `false` on a production cluster.
```
Comment (alicejw-aws) [inline _ml-commons-plugin/cluster-settings.md:15]:
```suggestion
If `plugins.ml_commons.only_run_on_ml_node` is set to`true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If its set to`false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as `false` on a production cluster. 
```

File: _ml-commons-plugin/cluster-settings.md [hunk 10]
```diff
@@ -47,49 +45,44 @@ plugins.ml_commons.task_dispatch_policy: round_robin
...
-When returning runtime information with the [profile API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api#profile), ML Commons will run a regular sync up job to sync up newly loaded or unloaded models on each node. When set to `0`, ML Commons immediately stops sync up jobs.
-
 ### Values
 
 - Default value: `10`
 - Value range: [0, 86,400]
 
 ## Predict monitoring requests
 
+Controls how many predict requests are monitored on one node. If set to `0`, OpenSearch clears all monitoring predict requests in the node's cache, and does not monitor predict requests from that point forward.
```
Comment (alicejw-aws) [inline _ml-commons-plugin/cluster-settings.md:63]:
```suggestion
The `monitoring_request_count ` setting controls how many predict requests are monitored on one node. If set to `0`, OpenSearch clears all monitoring predict requests in the node's cache, and does not monitor predict requests from that point forward.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 11]
```diff
@@ -7,37 +7,35 @@ nav_order: 10
...
-
 ### Values
 
-- Default value: `false`
+- Default value: `true`
 - Value range: `true` or `false`
 
 ## Dispatch tasks to ML node 
 
+`round_robin` dispatches ML tasks to ML nodes using round robin routing. `least_load` gathers all ML nodes' runtime information, such as JVM heap memory usage and running tasks, then dispatches tasks to the ML node with the least load.
```
Comment (alicejw-aws) [inline _ml-commons-plugin/cluster-settings.md:30]:
```suggestion
The `round_robin` setting dispatches ML tasks to ML nodes using round robin routing. `least_load` gathers all ML nodes' runtime information, such as JVM heap memory usage and running tasks, then dispatches tasks to the ML node with the least load.
```

File: _ml-commons-plugin/cluster-settings.md (hunk 10 above)
Comment (alicejw-aws) [inline _ml-commons-plugin/cluster-settings.md:63]: suggestion: 

"The `monitoring_request_count` setting controls how many. . ."

File: _ml-commons-plugin/cluster-settings.md [hunk 12]
```diff
@@ -47,49 +45,44 @@ plugins.ml_commons.task_dispatch_policy: round_robin
...
-Controls how many upload model tasks can run in parallel on one node. If set to `0`, you cannot upload models to any node.
-
 ### Value range
 
 - Default value: `100`
 - Value range: [0, 100,000,000]
 
 ## Upload model tasks per node
 
+Controls how many upload model tasks can run in parallel on one node. If set to `0`, you cannot upload models to any node.
```
Comment (alicejw-aws) [inline _ml-commons-plugin/cluster-settings.md:78]: suggestion

"The `max_upload_model_tasks_per_node` setting controls. . "

File: _ml-commons-plugin/cluster-settings.md [hunk 13]
```diff
@@ -99,33 +92,30 @@ Controls how many upload model tasks can run in parallel on one node. If set to
 
 ## Load model tasks per node
 
+Controls how many load model tasks can run in parallel on one node. If set to `0`, you cannot load models to any node.
```
Comment (alicejw-aws) [inline _ml-commons-plugin/cluster-settings.md:95]: similar suggestion, introduce setting by name.

**Review by alicejw-aws — APPROVED** [review]
Looks good!

**Comment by Naarcha-AWS** [comment]
Going to merge in for now and change the formatting to a table in a separate PR.

**Pull request merged**